### PR TITLE
fix: fix test cases and improve positions in errors

### PR DESCRIPTION
**Pull request by matt2e**
Changes
- Incomplete module boilerplate for test modules
- `ftl/one` was using `two.User`, but `ftl/two` was not including it in its schema
    - I've added a new Verb which makes use of User
- compiling schema was returning errors at the node that was reached before finding an error, where as sometimes we have a more accurate position available
    - Now we pass around the latest position we dive deeper into all the checks, rather than staying at the node's position
    - Previously we would get an error on the verb location, rather than the field on a struct that was used by the verb

Not included in this PR is a fix for #1095, which is why test/data is using Option in the new verb

### CODE REVIEW

File: go-runtime/compile/schema.go [hunk 1]
```diff
@@ -508,7 +508,7 @@ func visitComments(doc *ast.CommentGroup) []string {
 	return comments
 }
 
-func visitStruct(pctx *parseContext, node ast.Node, tnode types.Type) (*schema.Ref, error) {
+func visitStruct(pctx *parseContext, pos token.Pos, tnode types.Type) (*schema.Ref, error) {
 	named, ok := tnode.(*types.Named)
 	if !ok {
 		return nil, fmt.Errorf("expected named type but got %s", tnode)
```
Comment (alecthomas) [inline go-runtime/compile/schema.go:514]: The raw fmt.Errorf() calls in this function should now use errorf() if they can?

File: go-runtime/compile/schema.go [hunk 2]
```diff
@@ -686,90 +685,90 @@ func visitType(pctx *parseContext, node ast.Node, tnode types.Type) (schema.Type
...
-		return visitSlice(pctx, node, underlying)
+		return visitSlice(pctx, pos, underlying)
 
 	case *types.Interface:
 		if underlying.String() == "any" {
-			return &schema.Any{Pos: goPosToSchemaPos(node.Pos())}, nil
+			return &schema.Any{Pos: goPosToSchemaPos(pos)}, nil
 		}
-		return nil, errorf(node, "unsupported type %T", node)
+		return nil, errorf(pos, "unsupported type %T", tnode)
```
Comment (matt2e) [inline go-runtime/compile/schema.go:738]: Another change: this was printing the wrong variable before

File: go-runtime/compile/schema.go [hunk 3]
```diff
@@ -686,90 +685,90 @@ func visitType(pctx *parseContext, node ast.Node, tnode types.Type) (schema.Type
...
 
 	default:
-		return nil, errorf(node, "unsupported type %T", node)
+		return nil, errorf(pos, "unsupported type %T", tnode)
```
Comment (matt2e) [inline go-runtime/compile/schema.go:741]: Same variable mixup here as well

**Comment by matt2e** [comment]
Looking into why the tests are failing on github still

**Review by alecthomas — APPROVED** [review]

**Comment by matt2e** [comment]
> Looking into why the tests are failing on github still

I'm not able to repro on my machine... any ideas @alecthomas?

**Comment by alecthomas** [comment]
Nothing springs to mind. I'd try `git clean -fdx`, you might have some weird state lying around.

File: go-runtime/compile/schema_test.go [hunk 4]
```diff
@@ -13,10 +14,34 @@ import (
...
 	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/internal/exec"
+	"github.com/TBD54566975/ftl/internal/log"
 	"github.com/TBD54566975/ftl/internal/slices"
 )
 
+func prebuildTestModule(t *testing.T, args ...string) {
+	//this is helpful when a test requires another module to be built before running
+	//eg: when module A depends on module B, we need to build module B before building module A
+	ctx := log.ContextWithLogger(context.Background(), log.Configure(os.Stderr, log.Config{}))
```
Comment (alecthomas) [inline go-runtime/compile/schema_test.go:25]: Use `ContextWithNewDefaultLogger(context.Background())` instead.

File: go-runtime/compile/schema_test.go [hunk 5]
```diff
@@ -13,10 +14,34 @@ import (
...
+
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("Failed to get current directory: %v", err)
```
Comment (alecthomas) [inline go-runtime/compile/schema_test.go:29]: `assert.NoError(t, err)`

File: go-runtime/compile/schema_test.go [hunk 6]
```diff
@@ -13,10 +14,34 @@ import (
...
+	}
+
+	ftlArgs := []string{"build"}
+	ftlArgs = append(ftlArgs, args...)
+
+	cmd := exec.Command(ctx, log.Debug, dir, "ftl", ftlArgs...)
+	err = cmd.Run()
+	if err != nil {
```
Comment (alecthomas) [inline go-runtime/compile/schema_test.go:37]: `assert.NoError(t, erro)`

File: go-runtime/compile/schema_test.go [hunk 7]
```diff
@@ -13,10 +14,34 @@ import (
...
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("Failed to get current directory: %v", err)
+	}
+
+	ftlArgs := []string{"build"}
+	ftlArgs = append(ftlArgs, args...)
+
+	cmd := exec.Command(ctx, log.Debug, dir, "ftl", ftlArgs...)
+	err = cmd.Run()
```
Comment (alecthomas) [inline go-runtime/compile/schema_test.go:36]: Use `.RunBuffered()` instead.

**Review by alecthomas — APPROVED** [review]

**Pull request merged**